version: '3.3'

services:
  cratedb:
    image: crate:latest
    environment:
      - CRATE_HEAP_SIZE=${CRATE_HEAP_SIZE}
      - CRATE_NETWORK_HOST=${CRATE_NETWORK_HOST}
      - CRATE_PG_HOST=${CRATE_PG_HOST}
    volumes:
      - cratedb_data:/data
    networks:
      - crate-net
    expose:
      - "4200" 

  grafana:
    image: grafana/grafana
    environment:
      - GF_SECURITY_ADMIN_USER=${GRAFANA_ADMIN_USER}
      - GF_SECURITY_ADMIN_PASSWORD=${GRAFANA_ADMIN_PASSWORD}
      - GF_SERVER_ROOT_URL=${URL_GRAFANA}

    volumes:
      - grafana-data:/var/lib/grafana
    networks:
      - crate-net
    expose:
      - "3000"

  flask-api:
    build: ../API
    depends_on:
      - cratedb
    volumes:
      - ../API:/app
    env_file:
      - ../BD/.env
    networks:
      - crate-net
    expose:
      - "5000"

  data-generator:
    build: ../data_generator
    depends_on:
      - flask-api
    env_file:
      - ../BD/.env
    networks:
      - crate-net

  nginx:
    image: nginx:latest
    ports:
      - "8080:80"
    volumes:
      - ./nginx/nginx.conf:/etc/nginx/nginx.conf:ro
    depends_on:
      - cratedb
      - grafana
      - flask-api
    networks:
      - crate-net

volumes:
  cratedb_data: {}
  grafana-data: {}

networks:
  crate-net:
    external: true
